pkg/util/mount/mount_linux.go

```diff
@@ -31,28 +32,80 @@ import (
 	"github.com/golang/glog"
 )
 
-const FlagBind = syscall.MS_BIND
-const FlagReadOnly = syscall.MS_RDONLY
+const (
+	// How many times to retry for a consistent read of /proc/mounts.
+	maxListTries = 3
+	// Number of fields per line in "/proc/mounts", as per the fstab man page.
+	expectedNumFieldsPerLine = 6
+)
 
 // Mounter implements mount.Interface for linux platform.
 type Mounter struct{}
 
-// Mount wraps syscall.Mount()
-func (mounter *Mounter) Mount(source string, target string, fstype string, flags uintptr, data string) error {
-	glog.V(5).Infof("Mounting %s %s %s %d %s", source, target, fstype, flags, data)
-	return syscall.Mount(source, target, fstype, flags, data)
+// Mount mounts source to target as fstype with given options. 'source' and 'fstype' must
+// be an emtpy string in case it's not required, e.g. for remount, or for auto filesystem
+// type, where kernel handles fs type for you. The mount 'options' currently come from
+// mount(8), if no more option is required, call Mount with an empty string list or nil.
+func (mounter *Mounter) Mount(source string, target string, fstype string, options []string) error {
+	glog.V(5).Infof("Mounting %s %s %s %v", source, target, fstype, options)
+	// Build mount command as follows:
+	//   mount [-t $fstype] [-o $options] [$source] $target
+	mountArgs := []string{}
+	if len(fstype) > 0 {
+		mountArgs = append(mountArgs, "-t", fstype)
+	}
+	if options != nil {
+		// If options is an empty list, this is a no-op.
+		mountArgs = append(mountArgs, options...)
```

Comment:
on further reading, I guess you are expecting users to pass [--bind,  -o ro, -o remount] not just [bind,ro,remount] ?  Is that really the interface we want.  That's pretty highly coupled to the mount tool.

I'm not sure we really ought to expose the full range of stuff that mount offers without REALLY solid use cases.